src/intrinsics.h

```diff
@@ -152,21 +153,25 @@ extern "C"
 JL_CALLABLE(jl_f_intrinsic_call)
 {
     JL_NARGSV(intrinsic_call, 1);
-    JL_TYPECHK(intrinsic_call, intrinsic, args[0]);
-    enum intrinsic f = (enum intrinsic)*(uint32_t*)jl_data_ptr(args[0]);
+    JL_TYPECHK(intrinsic_call, intrinsic, F);
```

Comment:
Ideally we will stop using `JL_CALLABLE`, and instead use function-specific signatures that are more efficient where possible.